dashboard/app/views/levels/show.html.haml

```diff
@@ -22,6 +22,9 @@
   :javascript
     var appOptions = #{app_options.to_json};
   = render partial: 'levels/unplug', locals: {app: @game.app}
+- elsif @level.is_a?(Applab) && app_options['readonlyWorkspace'] && !app_options[:channel]
```

Comment:
Do you want to write a test that would catch things like that in the future
;) (should be pretty easy to write a rails controller test, I can help you
with that)

Argh what -- we should probably make that a HashWithIndifferentAccess.

On Thu, Nov 5, 2015 at 11:47 AM, David Bailey <notifications@github.com>
wrote:

> In dashboard/app/views/levels/show.html.haml
> <https://github.com/code-dot-org/code-dot-org/pull/4980#discussion_r44059619>
> :
>
> > @@ -22,6 +22,9 @@
> >    :javascript
> >      var appOptions = #{app_options.to_json};
> >    = render partial: 'levels/unplug', locals: {app: @game.app}
> > +- elsif @level.is_a?(Applab) && app_options['readonlyWorkspace'] && !app_options[:channel]
>
> perhaps it looked weird because my logic was wrong and would have totally
> broken the feature. Both of those fields are strings, so now I'm accessing
> both as strings. FWIW, app_options has quite a mix of string and symbol
> keys:
>
> 0> app_options.keys
> => [:levelGameName, :skinId, :baseUrl, :app, :levelId, :level, :cacheBust, :droplet, :pretty, :scriptId, :scriptName, "fullWidth", "smallFooter", "hasI18n", "callouts", "channel", "readonlyWorkspace", "isChannelBacked", "postMilestone", :applabUserId, :isAdmin, :isSignedIn, :pinWorkspaceToBottom, :hasVerticalScrollbars, :showExampleTestButtons, :rackEnv, :report, :sendToPhone, :send_to_phone_url, :copyrightStrings]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/4980/files#r44059619>.
>